test/graaflib/algorithm/coloring/welsh_powell_test.cpp

```diff
@@ -0,0 +1,165 @@
+#include <graaflib/algorithm/coloring/welsh_powell.h>
+#include <graaflib/graph.h>
+#include <graaflib/types.h>
+#include <gtest/gtest.h>
+
+#include <unordered_map>
+#include <unordered_set>
+#include <utility>
+#include <vector>
+
+namespace graaf::algorithm {
+
+namespace {
+
+template <typename T>
+struct TypedColoringTest : public testing::Test {
+  using graph_t = T;
+  using edge_t = typename T::second_type;
+};
+
+using graph_types =
+    testing::Types<directed_graph<int, int>, undirected_graph<int, int>>;
+
+TYPED_TEST_SUITE(GraphColoringTest, graph_types);
+
+}  // namespace
+
+template <typename T>
+struct WelshPowellTest : public testing::Test {
+  using graph_t = T;
+};
+
+using graph_types =
+    testing::Types<directed_graph<int, int>, undirected_graph<int, int>>;
+
+TYPED_TEST_SUITE(WelshPowellTest, graph_types);
+
+// Test case for an empty graph
+TYPED_TEST(WelshPowellTest , EmptyGraph) {
+  // GIVEN
+  using graph_t = typename TestFixture::graph_t;
+  graph_t graph{};
+
+  // WHEN
+  auto coloring = welsh_powell(graph);
```

Comment:
[typo]: same for the rest of this file

```suggestion
  // WHEN
  auto coloring = welsh_powell_coloring(graph);
```

Reply:
Done!!!